quiz-participant-client/src/quiz/QuizDashboard.tsx

```diff
@@ -36,6 +42,28 @@ const QuizDashboard: React.FC<QuizDashboardProps> = (props: QuizDashboardProps)
        return quiz.participants.some(p => p.turn && p.id === props.participantId)
     }
 
+    const isCurrentQuestionOpen = () => {
+        return !quiz.participants.some(p => p.turn);
+    }
+
+    useEffect(() => {
+        // Trigger preloading of audio to prevent delays when buzzer is pressed
+        buzzerAudio.current.muted = true;
+        buzzerAudio.current.play();
+    }, [])
+
+    useEffect(() => {
+        const buzzerOnKeydown = (event) => {
+            if ((event.keyCode === 32 || event.keyCode === 13) && isCurrentQuestionOpen()) {
+                buzzer();
+            }
+        }
+        document.addEventListener('keydown', buzzerOnKeydown);
+        return () => {
+            document.removeEventListener('keydown', buzzerOnKeydown);
+        }
+    }, []);
```

Comment:
Hier wird mir in der IDE folgende Warnung angezeigt: `React Hook useEffect has missing dependencies: 'buzzer' and 'isCurrentQuestionOpen'. Either include them or remove the dependency array.`
Ich würde den Buzzer mit seinen Hooks in eine eigene Komponente verschrieben. Das buzzern kollidiert sorgt im Moment gerade dafür, dass das ganze Dashboard neu gerendered und darum der Websocket geschlossen wird.